docs/examples/code/playwright_crawler.py

```diff
@@ -47,6 +48,13 @@ async def request_handler(context: PlaywrightCrawlingContext) -> None:
         # Find a link to the next page and enqueue it if it exists.
         await context.enqueue_links(selector='.morelink')
 
+    # Define a hook that will be called each time before navigating to a new URL.
+    # The hook receives a context parameter, providing access to the request among
+    # other things. In this example, we log the URL being navigated to.
+    @crawler.pre_navigation_hook
+    async def log_navigation_url(context: BasicCrawlingContext) -> None:
```

Comment:
While valid, `BasicCrawlingContext` does not provide `page` - let's update this to `BrowserPageContext`.

Reply:
you mean `BrowserCrawlingContext` i guess? that's how its called in the JS version

Reply:
We don't have a `BrowserCrawlingContext` here. And `PlaywrightCrawlingContext` contains stuff that doesn't make sense in a pre-navigation hook, like `infinite_scroll`.

Reply:
well, it will contain things that are valid once we implement them, like `blockRequests` helper

you get pw context there, so the type should reflect that

Reply:
I don't think we are in disagreement here. There is an inheritance chain like this: `BasicCrawlingContext` <- `BrowserPageContext` <- `PlaywrightCrawlingContext`. `BrowserPageContext` gets passed to pre-navigation hooks, `PlaywrightCrawlingContext` goes into the request handler.

If you're suggesting that `BrowserPageContext` could have a better name, that's a good point.

Reply:
@Prathamesh010 better, but still not optimal... I'd either name it after the intended usage (in prenavigation hooks) or just try to convey the fact that there's less stuff than in regular `PlaywrightCrawlingContext`.

Reply:
afaik in JS we have this typed to `PlaywrightCrawlingContext`, do we really pass something else in the hooks? `PlaywrightCrawlingContext` is where we add the context helpers, so it should be that one, its not just about having the `page` property, its about the context helpers. 

i understand that some helpers might not make sense in here (maybe even all of these we currently have), and i am fine with dealing with this later (once we have some helpers that should be here, like the `blockRequests`)

the JS version has the browser crawler abstraction, not sure if its really needed here, given we will have just one browser crawler type (no puppeteer). i can imagine we will add the CDP crawler here too at some point, but that is a rather distant future

Reply:
In JS, we mutate the context as it proceeds through the request handling process, so we promise the complete `PlaywrightCrawlingContext` in the `PlaywrightCrawler` pre-navigation hooks. In the Python port, we are trying to only pass whatever makes sense - hence the intermediate type.

Reply:
got it (and I surely like it too), so let's just try to find a better-suited name

Reply:
out of those, I'd say `PreNavigationContext` is the best one